RFS/src/main/java/org/opensearch/migrations/bulkload/version_universal/RemoteReaderClient.java

```diff
@@ -121,8 +121,8 @@ Mono<ObjectNode> getJsonForTemplateApis(HttpResponse resp) {
         try {
             var tree = (ObjectNode) objectMapper.readTree(resp.body);
 
-            if (tree.size() == 1) {
-                return Mono.just(handleSingleItemTree(tree));
+            if (tree.size() == 1 && tree.fields().next().getValue().isArray()) {
```

Comment:
Whoa, what broke here?  From these changes the shape of the response is different between the OSS and general product line?

Reply:
This is a good candidate to adding a unit test, LMK if you'd prefer I bootstrap these tests.

Reply:
The shape isn't different between oss and non-oss, it's just that we previously weren't testing the case of 6.8 with only one item

Getting:
```
{"simple_index_template":{"order":0,"index_patterns":["blog*"],"settings":{"index":{"number_of_shards":"1"}},"mappings":{"_doc":{"_source":{"enabled":true},"properties":{"created_at":{"format":"EEE MMM dd HH:mm:ss Z yyyy","type":"date"},"host_name":{"type":"keyword"}}}},"aliases":{"alias1":{}}}}
```

Whereas 7.10 gives
```
{"component_templates":[{"name":"simple_component_template","component_template":{"template":{"settings":{"index":{"number_of_shards":"1","number_of_replicas":"1"}},"mappings":{"properties":{"author":{"type":"text"}}},"aliases":{"alias1":{}}},"version":1}}]}
```

Before, with non-oss we were hitting the case where size was > 1

Reply:
Will merge and follow up if any unit tests can be added